_pages/clients/r.md

```diff
@@ -56,207 +54,219 @@ If you are running [Algorithmia Enterprise](/enterprise), you can specify the AP
 
-Alternately, if you just need the text content of the file to be stored in a variable, you can retrieve the remote file's content without saving the actual file:
+### Create a Data Collection
+
+The [documentation for nlp/CleanDocuments](https://algorithmia.com/algorithms/nlp/CleanDocuments/docs) says that it takes a URL or a Data URI of the documents to be processed, and returns a Data URI where the result can be downloaded. We'll create a directory to host the input documents, then update its [permissions](/developers/api/#update-collection-acl) so that it's publicly accessible:
 
 {% highlight r %}
-# Download contents of file as a string
-if (client$file(text_file)$exists()) {
-  input <- client$file(text_file)$getString()
+# Instantiate a Data Directory object
+nlp_directory = client$dir("data://YOUR_USERNAME/nlp_directory")
+# Create your data collection if it does not exist
+if(nlp_directory.exists() == FALSE) {
+    nlp_directory.create()
 }
+# Change permissions on your data collection to public
+nlp_directory$update_permissions(ReadAcl.PUBLIC)
 {% endhighlight %}
 
-This will get your file as a string, saving it to the variable `input`.  If the file was binary (an image, etc), you could instead use the function `getBytes()` to retrieve the file's content as a byte array. For more image-manipulation tutorials, see the [Computer Vision Recipes]({{site.baseurl}}/tutorials/recipes/#computer-vision).
+Instead of your username you can also use '.my' when calling algorithms. For more information about the '.my' pseudonym check out the [Hosted Data Guide]({{site.baseurl}}/data/hosted).
+{: .notice-info}
 
-Now you've seen how to upload a local data file, check if a file exists in a data collection, and download the file contents.
+### Upload Data to your Data Collection
 
-For more methods on how to get a file using the Data API from a data collection go to the [API Specification](http://docs.algorithmia.com/#getting-a-file).
+Now we're ready to upload an text document for processing. For this example, we'll use the first chapter of Jack London's [Burning Daylight](https://en.wikisource.org/wiki/Burning_Daylight). Since the "Clean Documents" algorithm can process multiple documents in the same text file, it requires each document to be separated by the delimiter `"""`. The algorithm will throw an error if the delimimter isn't present. You can download the first chapter with the delimiter already added here: [Chapter One Burning Daylight, by Jack London](/data_assets/burning_daylight_delimiter.txt).
```

Comment:
```suggestion
Now we're ready to upload an text document for processing. For this example, we'll use the first chapter of Jack London's [Burning Daylight](https://en.wikisource.org/wiki/Burning_Daylight). Since the "Clean Documents" algorithm can process multiple documents in the same text file, it requires each document to be separated by the delimiter `"""`. The algorithm will throw an error if the delimiter isn't present. You can download the first chapter with the delimiter already added here: [Chapter One Burning Daylight, by Jack London](/data_assets/burning_daylight_delimiter.txt).
```